tests/contest/contest/src/tests/linux_masked_paths/masked_paths.rs

```diff
@@ -0,0 +1,227 @@
+            io::Result::Err(e) => {
+                bail!("error in creating symlink, to {:?} {:?}", test_file, e);
+            }
+        }
+
+        let r_path = match fs::read_link(&test_file) {
+            io::Result::Ok(p) => p,
+            io::Result::Err(e) => {
+                bail!("error in reading symlink at {:?} : {:?}", test_file, e);
+            }
+        };
+
+        // It ensures that the symlink points not to exist.
+        match fs::metadata(r_path) {
+            io::Result::Ok(md) => {
+                bail!(
+                    "reading path {:?} should have given error, found {:?} instead",
+                    test_file,
+                    md
+                )
+            }
+            io::Result::Err(e) => {
+                let err = e.kind();
+                if let io::ErrorKind::NotFound = err {
+                    Ok(())
+                } else {
+                    bail!("expected not found error, got {:?}", err);
+                }
+            }
+        }
+    });
+
+    // If the container creation succeeds, we expect an error since the masked paths does not support symlinks.
+    if let TestResult::Passed = res {
+        TestResult::Failed(anyhow!(
+            "expected error in container creation with invalid symlink, found no error"
+        ))
+    } else {
+        TestResult::Passed
+    }
+}
+
+fn test_mode(mode: u32) -> TestResult {
+    const MASKED_DEVICE: &str = "masked_device";
+
+    let root = PathBuf::from("/");
+    let masked_paths = vec![root.join(MASKED_DEVICE)];
+    let spec = get_spec(masked_paths);
+
+    test_inside_container(spec, &CreateOptions::default(), &|bundle_path| {
+        use std::os::unix::fs::OpenOptionsExt;
+        use std::{fs, io};
+        let test_file = bundle_path.join(MASKED_DEVICE);
+
+        if let io::Result::Err(e) = fs::OpenOptions::new()
+            .mode(mode)
+            .create(true)
+            .write(true)
+            .open(&test_file)
+        {
+            bail!(
+                "could not create file {:?} with mode {:?} : {:?}",
+                test_file,
+                mode ^ 0o666,
+                e
+            );
+        }
+
+        match fs::metadata(&test_file) {
+            io::Result::Ok(_) => Ok(()),
+            io::Result::Err(e) => {
+                let err = e.kind();
+                if let io::ErrorKind::NotFound = err {
+                    bail!("error in creating device node, {:?}", e)
+                } else {
+                    Ok(())
+                }
+            }
+        }
+    })
+}
+
+fn check_masked_device_nodes() -> TestResult {
+    let modes = [
+        SFlag::S_IFBLK.bits() | 0o666,
+        SFlag::S_IFCHR.bits() | 0o666,
+        SFlag::S_IFIFO.bits() | 0o666,
+    ];
+    for mode in modes {
+        let res = test_mode(mode);
+        if let TestResult::Failed(_) = res {
+            return res;
+        }
+    }
+    TestResult::Passed
```

Comment:
```suggestion
    [
        SFlag::S_IFBLK.bits() | 0o666,
        SFlag::S_IFCHR.bits() | 0o666,
        SFlag::S_IFIFO.bits() | 0o666,
    ]
    .iter()
    .map(|mode| test_mode(*mode))
    .find(|res| {
        if let TestResult::Failed(_) = res {
            true
        } else {
            false
        }
    })
    .unwrap_or(TestResult::Passed)
```

Ideally, we should return the failure result of all modes.

Reply:
I fixed it in commit [5d79a48](https://github.com/youki-dev/youki/pull/2950/commits/5d79a486a3b67a977f1c789723424910bbd0b6ee).